src/main/host/syscall_condition.c

```diff
@@ -33,7 +33,7 @@ struct _SysCallCondition {
     // The thread waiting for the signal
     Thread* thread;
     // If a task to deliver a signal has been scheduled
-    bool signalPending;
+    bool signalScheduled;
```

Comment:
I think there are other parts of the code (in epoll maybe?) where "scheduled" is a state that turns back to false when the task later comes out of the event queue and is being run. Might be useful to either rename the bool, or add a comment to explain that once it becomes true we never set it to false again, or both.